pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+        """
+        if self.data['method'] == 'PolyScan' and len(self.data['params']) != 3:
+            raise ValueError('For PolyScan method, set params = order, number of '
+                             'pixels, number of repeats')
+
+        if self.data['method'] == 'bspline' and not isinstance(self.data['params'], int):
+            raise ValueError('For bspline method, set params = spacing (integer).')
+            
+        if self.data['frame'] in FlatFieldPar.valid_frames() or self.data['frame'] is None:
+            return
+
+        if not os.path.isfile(self.data['frame']):
+            raise ValueError('Provided frame file name does not exist: {0}'.format(
+                                self.data['frame']))
+
+
+class FlexurePar(ParSet):
+    def __init__(self, method=None, maxshift=None, spectrum=None):
+
+        # Grab the parameter names and values from the function
+        # arguments
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        pars = dict([(k,values[k]) for k in args[1:]])
+
+        # Initialize the other used specifications for this parameter
+        # set
+        defaults = dict.fromkeys(pars.keys())
+        options = dict.fromkeys(pars.keys())
+        dtypes = dict.fromkeys(pars.keys())
+        descr = dict.fromkeys(pars.keys())
+
+        # Fill out parameter specifications.  Only the values that are
+        # *not* None (i.e., the ones that are defined) need to be set
+
+        defaults['method'] = 'boxcar'
+        options['method'] = FlexurePar.valid_methods()
+        dtypes['method'] = str
+        descr['method'] = 'Method used to correct for flexure. Use None for no correction.  If ' \
+                          'slitcen is used, the flexure correction is performed before the ' \
+                          'extraction of objects.  ' \
+                          'Options are: None, {0}'.format(', '.join(options['method']))
+
+        defaults['maxshift'] = 20
+        dtypes['maxshift'] = [int, float]
+        descr['maxshift'] = 'Maximum allowed flexure shift in pixels.'
+
+        dtypes['spectrum'] = str
+        descr['spectrum'] = 'Archive sky spectrum to be used for the flexure correction.'
+
+        # Instantiate the parameter set
+        super(FlexurePar, self).__init__(list(pars.keys()),
+                                         values=list(pars.values()),
+                                         defaults=list(defaults.values()),
+                                         options=list(options.values()),
+                                         dtypes=list(dtypes.values()),
+                                         descr=list(descr.values()))
+        self.validate()
+
+    @classmethod
+    def from_dict(cls, cfg):
+        k = cfg.keys()
+        parkeys = [ 'method', 'maxshift', 'spectrum' ]
+        kwargs = {}
+        for pk in parkeys:
+            kwargs[pk] = cfg[pk] if pk in k else None
+        return cls(**kwargs)
+
+    @staticmethod
```

Comment:
These should be removed, right?

Reply:
As Joe also noted, these parameter sets need to be updated to whatever's current.  Also a integration/hackday issue...